### PR TITLE
Add missing Model methods, update AdminSite.get_app_list() args

### LINKED ISSUE #1405: Missing app_label for AdminSite.get_app_list in stub
# Bug report

## What's wrong
`django.contrib.admin.AdminSite.get_app_list` stub is missing the `app_label` argument.

Django's method:
```py
    def get_app_list(self, request, app_label=None):
        """
        Return a sorted list of all the installed apps that have been
        registered in this site.
        """
        app_dict = self._build_app_dict(request, app_label)

        # Sort the apps alphabetically.
        app_list = sorted(app_dict.values(), key=lambda x: x["name"].lower())

        # Sort the models alphabetically within each app.
        for app in app_list:
            app["models"].sort(key=lambda x: x["name"])

        return app_list
```
Django-stubs stub:
```py
def get_app_list(self, request: HttpRequest) -> list[Any]: ...
```

## How is that should be
```py
def get_app_list(self, request: HttpRequest, app_label: Optional[str]) -> list[Any]: ...
```

## System information

- `django-stubs` version: `1.15.0`

### PR DESCRIPTION
**Pull request by RyanWalker277**
# Refs

Fixes #1405 

# Changes

Added stubs for get_constraints, serializable_value, prepare_database_save and date_error_message. Also added `app_label` parameter in `get_app_list`

### CODE REVIEW

File: django-stubs/db/models/base.pyi [hunk 1]
```diff
@@ -46,8 +47,10 @@ class Model(metaclass=ModelBase):
     def clean(self) -> None: ...
     def clean_fields(self, exclude: Collection[str] | None = ...) -> None: ...
     def validate_unique(self, exclude: Collection[str] | None = ...) -> None: ...
+    def date_error_message(self, lookup_type: str, field_name: str, unique_for: str) -> ValidationError: ...
     def unique_error_message(self, model_class: type[Self], unique_check: Sequence[str]) -> ValidationError: ...
     def validate_constraints(self, exclude: Collection[str] | None = ...) -> None: ...
+    def get_constraints(self) -> list[tuple[type[Model], Sequence[CheckConstraint | UniqueConstraint]]]: ...
```
Comment (sobolevn) [inline django-stubs/db/models/base.pyi]:
```suggestion
    def get_constraints(self) -> list[tuple[type[Model], Sequence[BaseConstraint]]]: ...
```

**Review by sobolevn — COMMENTED** [review]
Thanks, one suggestion

**Comment by RyanWalker277** [comment]
> Thanks, one suggestion

Thanks for the help. :smile:

**Review by sobolevn — APPROVED** [review]

**Comment by RyanWalker277** [comment]
I have added one more change to fix #1405.

**Comment by sobolevn** [comment]
Next time, please create separate PRs for unrelated changes, this way it is easier to review and track changes, revert them if needed.

Thank you for your help!

**Pull request merged**